src/driving_distance/v6withPoints_dd_driver.cpp

```diff
@@ -0,0 +1,224 @@
+/*PGR-GNU*****************************************************************
+File: withPoints_dd_driver.cpp
+
+Generated with Template by:
+Copyright (c) 2015 pgRouting developers
+Mail: project@pgrouting.org
+
+Function's developer:
+Copyright (c) 2015 Celia Virginia Vergara Castillo
+Mail: vicky at erosion.dev
+
+Copyright (c) 2023 Yige Huang
+Mail: square1ge at gmail.com
+------
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+ ********************************************************************PGR-GNU*/
+
+
+#include "drivers/driving_distance/v6withPoints_dd_driver.h"
+
+#include <sstream>
+#include <deque>
+#include <vector>
+#include <algorithm>
+
+#include "dijkstra/drivingDist.hpp"
+#include "withPoints/pgr_withPoints.hpp"
+
+
+#include "cpp_common/pgr_alloc.hpp"
+
+
+char
+estimate_drivingSide_dd(char driving_side, bool directed, char** err_msg){
```

Comment:
Here is the function I wrote to check the driving side. I am inexperienced in error checking, can you please give me some advice？ @krashish8

Reply:
Last meeting, vicky says that I can make the check on SQL level or C/C++ level. But since I don't know how to check on the SQL level, I chose the latter. Thank you for your guidance and I will follow the PLPGSQL documentation and also the given example, trying to raise error on SQL level.

Actually, there will not any default value for `driving side` because I'm creating a new function that the `driving side` is compulsory rather than optional. User must give a proper value for `driving side` or the function will not work. Since there are some restrictions, like "Both" is only valid on undirected graphs (If `b` is given meanwhile graph is directed, I consider it should raise error because you cannot drive on both side in a directed graph), I think the check is needed.

Again, thanks for your guidance！